core/src/main/java/org/bitcoinj/script/Script.java

```diff
@@ -276,6 +276,27 @@ else if (isPayToScriptHash())
             throw new ScriptException("Script not in the standard scriptPubKey form");
     }
 
+    /**
+     * <p>If a program matches the standard template DUP HASH160 &lt;pubkey hash&gt; EQUALVERIFY CHECKSIG
+     * then this function retrieves the third element.
+     * In this case, this is useful for fetching the destination address of a transaction.</p>
+     *
+     * <p>If a program matches the standard template HASH160 &lt;script hash&gt; EQUAL
+     * then this function retrieves the second element.
+     * In this case, this is useful for fetching the hash of the redeem script of a transaction.</p>
+     *
+     * <p>Otherwise it return null as Exception creation is not a cheap process.</p>
+     *
+     */
+    public byte[] getPubKeyHashUnsafe() {
+        if (isSentToAddress())
+            return chunks.get(2).data;
+        else if (isPayToScriptHash())
+            return chunks.get(1).data;
+        else
+            return null;
```

Comment:
Instead of creating a new method, we could speed up the Exception-throwing in the other method using static Exceptions without stacktrace.